File: app/src/main/java/com/esoxjem/movieguide/di/modules/AppModule.java
package com.esoxjem.movieguide.di.modules;

import android.content.Context;
import android.content.res.Resources;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author arunsasidharan
 * @author pulkitkumar
 */
@Module
public class AppModule {

    @Provides
    @Singleton
    public Resources provideResources(Context context) {
        return context.getResources();
    }
}

File: app/src/main/java/com/esoxjem/movieguide/di/modules/AppModule.kt
package com.esoxjem.movieguide.di.modules

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author arunsasidharan
 * @author pulkitkumar
 */
@Module
class AppModule(application: Application) {

    private val context: Context

    init {
        context = application
    }

    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    fun provideSharedPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }

    companion object {
        private val PREF_NAME = "FavoritesStore"
    }
}
